ci/treefmt: add yamlfmt

wolfgangwalther opened the pull request:
Most workflow files are already well formatted, but to make it easier to keep it that way, we can add yamlfmt.

I personally have a preference for non-indented arrays for YAML, but wanted to avoid bigger diffs here - the status-quo clearly are indented arrays.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wolfgangwalther committed:
ci/treefmt: add yamlfmt

Most workflow files are already well formatted, but to make it easier to
keep it that way, we can add yamlfmt.

I personally have a preference for non-indented arrays for YAML, but
wanted to avoid bigger diffs here - the status-quo clearly are indented
arrays.

Some changes are made manually to the get-merge-commit action and the
issue templates. Those would otherwise make yamlfmt misbehave on those.

philiptaron commented:
I'm recusing myself from this review because it involves [thinking about the format of YAML](https://ruudvanasseldonk.com/2023/01/11/the-yaml-document-from-hell).

SigmaSquadron reviewed (commented):
If at all possible, I'd like to see two commits: one for the treewide format, and another that actually adds the `yamlfmt` configuration to `treefmt-nix`

wolfgangwalther committed:
treewide: run yamlfmt

This applies changes from running yamlfmt with the `retain_line_breaks`
formatting option enabled.

SigmaSquadron commented on .github/ISSUE_TEMPLATE/01_bug_report.yml:
```diff
@@ -101,7 +101,7 @@ body:
       description: |
         Please mention the people who are in the **Maintainers** list of the offending package. This is done by by searching for the package on the [NixOS Package Search](https://search.nixos.org/packages) and mentioning the people listed under **Maintainers** by prefixing their GitHub usernames with an '@' character. Please add the mentions above the `---` characters in the template below.
-      value: |
+      value: |2
```
Won't this add a random `2` to the actual template?

philiptaron reviewed (approved):
I said I was recusing myself but evidently I can hold my nose and read about YAML on a Friday.

https://yaml-multiline.info/ is cursed knowledge but I guess not _totally_ cursed.
